### PR TITLE
Transpose

### PR DESCRIPTION
**Pull request by RobinCsl**
There is a commit comment relating to files which should have been in their own branch, but I worked on master first, and then realised that to make a second PR, I should have used a separate branch... Hopefully that's not a problem!

### CODE REVIEW

**Comment by matthewmorgan** [comment]
@RobinCsl no worries at all about working on master-- since you're on your own fork, it's not a problem.  I'll review.

File: exercises/transpose/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Making the Test Suite Pass
+
+Execute the tests with:
+
+```bash
+$ npm test
+```
+
+In many test suites all but the first test have been skipped.
```
Comment (matthewmorgan) [inline exercises/transpose/README.md]: Same language issue I mentioned on your other PR.  All tests _have_ been skipped except the first, so might as well say so.

File: exercises/transpose/example.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export default function transpose(text) {
+  const result = [];
+  text.forEach((line, lineNo) => {
+    line.split('').forEach((value, key) => {
```
Comment (matthewmorgan) [inline exercises/transpose/example.js]:
```javascript
   [...line].forEach((value, key) => {
```

The `...` is the "spread" operator.  [It's awesome](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator).

File: exercises/transpose/example.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+      if (typeof result[key] === 'undefined') {
+        result[key] = ' '.repeat(lineNo);
+      }
+      result[key] += value;
+    });
+  });
+  return result;
+}
```
Comment (matthewmorgan) [inline exercises/transpose/example.js]: This all looks good, but it did make me wonder if it's possible to do this entirely functionally-- IE, without having an initial `result` array that gets mutated.  Not critical, but just thinking out loud...

**Comment by RobinCsl** [comment]
I updated the README and used the spread operator (it's really cool indeed!). I tried briefly to think about rewriting the example in a purely functional way, but to no avail at the moment.

**Comment by matthewmorgan** [comment]
Looks fine to me!  Mind squashing the commits?

**Comment by RobinCsl** [comment]
It seems I somehow managed on this one. I'll try to replicate the steps. (But there might be other conflicts since each exercise implementation modifies config.json)

**Comment by rchavarria** [comment]
@matthewmorgan approved changes, commits squased, no conflicts,... merging.

**Comment by shybyte** [comment]
@RobinCsl The expected test result in the last test ("test many lines") is different from the expected test result in https://github.com/exercism/problem-specifications/blob/master/exercises/transpose/canonical-data.json 

So either the canonical test-data is wrong or this PR is wrong.

**Comment by matthewmorgan** [comment]
@shybyte good catch! @RobinCsl I need to remove this exercise for the time being.  Please have a look and let me know if I can help resolving the issue.

**Comment by RobinCsl** [comment]
I had a look for the last hour. I had copy-pasted the code from the old repository and converted it to ES6, but you're right @shybyte, there is an issue with the algorithm. I will work more on that.

**Comment by shybyte** [comment]
Maybe my code for solving the original testcases is helpful:

```javascript
function trimTrailingUndefined(array) {
  const trailingUndefinedCount = [...array].reverse().findIndex(x => x !== undefined);
  return array.slice(0, array.length - trailingUndefinedCount);
}

export default function transpose(input) {
  const maxCol = Math.max(0, ...(input.map(row => row.length)));
  return [...Array(maxCol).keys()].map(col =>
    trimTrailingUndefined(input.map((_v, row) => input[row][col]))
      .map(charOrUndefined => charOrUndefined || ' ')
      .join('')
  );
}
```

**Pull request merged**